src/vs/workbench/services/themes/browser/workbenchThemeService.ts

```diff
@@ -447,6 +447,27 @@ export class WorkbenchThemeService implements IWorkbenchThemeService {
 		return this.onColorThemeChange.event;
 	}
 
+	public async togglePreferredTheme(): Promise<IWorkbenchColorTheme | null> {
```

Comment:
I try to keep the API of the `IWorkbenchThemeService` as minimal as possible so I'd rather not add an additional API to it.
Instead, the code should go into the command implementation. I know that's a bit of duplication, but the code is simple. `(await service.getColorThemes()).some(t => t.settingsId == settingsId)` to find a theme and service.setColorTheme(.., 'auto') to apply the theme.

Reply:
Without passing from `IWorkbenchThemeService` where can I find the theme for a given `settingsId`?

Reply:
Find a theme for the given settingsId: 
`(await service.getColorThemes()).some(t => t.settingsId == settingsId)`

get the value of a `setting = IConfigurationService.getValue(ThemeSettings.PREFERRED_DARK_THEME)`

Reply:
I think you meant `.find` instead of `.some`, right?

Thank you very much for all the hints, I pushed a new commit!